drivers/interrupt_controller/intc_renesas_ra_icu.c

```diff
@@ -26,99 +30,112 @@ enum {
 	IELSRn_OFFSET = 0x300,
 };
 
-int ra_icu_query_exists_irq(uint32_t event)
+static inline unsigned int event_to_irq(unsigned int event)
 {
-	for (uint32_t i = 0; i < CONFIG_NUM_IRQS; i++) {
-		uint32_t els = sys_read32(IELSRn_REG(i)) & UINT8_MAX;
-
-		if (event == els) {
-			return i;
-		}
+	if (event >= sizeof(function_irq_table)) {
+		return -1;
 	}
 
-	return -EINVAL;
+	return function_irq_table[event];
+}
+
+#ifdef CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER
+void arm_irq_priority_set(unsigned int irq, unsigned int prio, uint32_t flags);
+
+void z_soc_irq_init(void)
+{
 }
 
-int ra_icu_query_available_irq(uint32_t event)
+void z_soc_irq_enable(unsigned int event)
 {
-	int irq = -EINVAL;
+	const unsigned int irq = event_to_irq(event);
 
```

Comment:
Why do you mixing cons and non-const here?

Reply:
The constness of a pass-by-value return value is determined only by the const specification of the left-hand side, whether const or not of the right-hand side.
Here, const is specified to ensure that the compiler checks unintended changes and for readability. In some cases, it may also be a hint for compiler optimization.

Reply:
But in the same time, you are mixing signed/unsigned and drop the errno....